drivers/vfio/pci/nvgrace-gpu/egm.c

```diff
@@ -368,11 +371,16 @@ int register_egm_node(struct pci_dev *pdev)
 
 	nvgrace_egm_fetch_bad_pages(pdev, region);
 
-	list_add_tail(&region->list, &egm_list);
+	ret = setup_egm_chardev(region);
+	if (ret)
+		goto err;
 
-	setup_egm_chardev(region);
+	list_add_tail(&region->list, &egm_list);
```

Comment:
Why change the order for list_add_tail and setup_egm_chardev?

* With the new order, it doesn't affect the setup_egm_chardev.

Reply:
make sense